thinkhazard_processing/processing.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+import transaction
+import datetime
+import numpy
+import rasterio
+import pyproj
+from rasterio import features
+from shapely.ops import transform
+from functools import partial
+from geoalchemy2.shape import to_shape
+
+from thinkhazard_common.models import (
+    DBSession,
+    AdministrativeDivision,
+    HazardLevel,
+    )
+from .models import (
+    HazardSet,
+    Layer,
+    Output,
+    )
+from . import settings
+
+
+class ProcessException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+def process_pendings():
+    ids = DBSession.query(HazardSet.id) \
+        .filter(HazardSet.complete.is_(True)) \
+        .filter(HazardSet.processed.is_(False))
+    for id in ids:
+        process_hazardset(id)
+
+
+def process_all():
+    ids = DBSession.query(HazardSet.id) \
+        .filter(HazardSet.complete.is_(True))
+    for id in ids:
+        process_hazardset(id, force=True)
+
+
+def process_hazardset(hazardset_id, force=False):
+    print hazardset_id
+    chrono = datetime.datetime.now()
+
+    hazardset = DBSession.query(HazardSet).get(hazardset_id)
+    if hazardset is None:
+        raise ProcessException('HazardSet {} does not exist.'
+                               .format(hazardset_id))
+
+    if hazardset.processed:
+        if force:
+            hazardset.processed = False
+            DBSession.flush()
+        else:
+            raise ProcessException('HazardSet {} has already been processed.'
+                                   .format(hazardset_id))
+
+    # lean previous outputs
+    DBSession.query(Output) \
+        .filter(hazardset_id == hazardset_id) \
+        .delete()
+
+    hazardtype = hazardset.hazardtype
+    hazardtype_settings = settings['hazard_types'][hazardtype.mnemonic]
+    threshold = hazardtype_settings['threshold']
+
+    project = partial(pyproj.transform,
```

Comment:
It's actually how it is defined in shapely documentation. I don't like it much but wouldn't change it though.